klein-consensus/klein-consensus-paxos/src/main/java/com/ofcoder/klein/consensus/paxos/core/ProxyProposer.java

```diff
@@ -0,0 +1,56 @@
+package com.ofcoder.klein.consensus.paxos.core;
+
+import com.google.common.collect.Lists;
+import com.ofcoder.klein.common.exception.RedirectException;
+import com.ofcoder.klein.consensus.facade.AbstractInvokeCallback;
+import com.ofcoder.klein.consensus.facade.Result;
+import com.ofcoder.klein.consensus.paxos.PaxosNode;
+import com.ofcoder.klein.consensus.paxos.Proposal;
+import com.ofcoder.klein.consensus.paxos.rpc.vo.RedirectReq;
+import com.ofcoder.klein.consensus.paxos.rpc.vo.RedirectRes;
+import com.ofcoder.klein.rpc.facade.Endpoint;
+import com.ofcoder.klein.rpc.facade.RpcClient;
+import com.ofcoder.klein.spi.ExtensionLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * @author petter
+ */
+public class ProxyProposer implements ProposerRedirect{
+
+    private static final Logger LOG = LoggerFactory.getLogger(ProxyProposer.class);
+    private RpcClient client;
+
+    public ProxyProposer() {
+        this.client = ExtensionLoader.getExtensionLoader(RpcClient.class).getJoin();
+    }
+
+    @Override
+    public <E extends Serializable> void call(String group, E data, PaxosNode node, ProposeDone done) {
+        Endpoint master = node.getMemberConfiguration().getMaster();
```

Comment:
here, u should determine whether the master is itself . if is itself, it should directly call Proposer. Otherwise, it needs to  RPC to minimize RPC times